client/lxd_instances.go

```diff
@@ -1134,6 +1135,52 @@ func (r *ProtocolLXD) ExecInstance(instanceName string, exec api.InstanceExecPos
 			}
 		}
 
+		if exec.RecordOutput && (args.Stdout != nil || args.Stderr != nil) {
+			err = op.Wait()
+			if err != nil {
+				return nil, err
+			}
+
+			opAPI = op.Get()
+			outputFiles := map[string]string{}
+			outputs, ok := opAPI.Metadata["output"].(map[string]any)
+			if ok {
+				for k, v := range outputs {
+					outputFiles[k] = v.(string)
+				}
+			}
+
+			if outputFiles["1"] != "" {
+				reader, _ := r.getInstanceExecOutputLogFile(instanceName, filepath.Base(outputFiles["1"]))
+				if args.Stdout != nil {
+					_, err = io.Copy(args.Stdout, reader)
+					if err != nil {
+						return nil, fmt.Errorf("Could not copy the content of %q exec output log file to stdout: %v", instanceName, err)
```

Comment:
If the copy fails we should still attempt to delete the file before returning otherwise we'll leave files on the lxd server.